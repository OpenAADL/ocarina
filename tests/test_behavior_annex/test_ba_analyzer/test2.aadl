package exemple2
public
   with Base_Types ;

thread th
end th;


thread implementation th.good

annex behavior_specification {**

states
 uniqueState : initial final state ;
 finalState1 : final state ;
 compState : complete state ; 
  
**};

end th.good;

thread implementation th.error 
annex behavior_specification {**

  -- Errors: The behavior annex of a thread
  --          _ can't have more than one initial state.
  --          _ must define one or more final states.
  --          _ must define one or more complete states. 
states
 initState1 : initial state ;
 initState2 : initial state ;
 
**};
end th.error;

end exemple2 ;
