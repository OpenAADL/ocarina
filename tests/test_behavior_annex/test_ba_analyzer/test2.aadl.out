test2.aadl:31:02: th.error ( Thread ) can't have more than one initial state.
test2.aadl:29:01: th.error ( Thread ) must define one or more complete states.
test2.aadl:29:01: th.error ( Thread ) must define one or more final states.
Cannot analyze BEHAVIOR_SPECIFICATION specifications
